contracts/ERC721M.sol

```diff
@@ -31,6 +33,8 @@ contract ERC721M is IERC721M, ERC721AQueryable, Ownable, ReentrancyGuard {
         private _stageMintedCountsPerWallet;
     mapping(uint256 => uint256) private _stageMintedCounts;
 
+    mapping(address => Counters.Counter) private _cosignNonces;
```

Comment:
just use the minted count as the nonce, so we can save the storage of this mapping here